zerver/views/events_register.py

```diff
@@ -110,6 +111,7 @@ def events_register_backend(
     #       but we will still need to support tuples for a long time.
     modern_narrow = narrow_dataclasses_from_tuples(narrow)
 
+    partial_users = os.environ.get("PARTIAL_USERS") is not None
```

Comment:
A possible way for this to work might be to instead have this live in `get_user_dicts_in_realm`, and just have it make the choice to send the full user set when `PARTIAL_USERS` if False, preserving the legacy behavior? The API design we've specified will allow the server to send the full user set at its discretion, so I think that'd be a nice approach.